### PR TITLE
Proposal for Workflows

### PR DESCRIPTION
**Pull request by massich**
This PR is a playground reorganizing some MNE-documentaiton: 
(I'll rewrite the PR content at some point, to make our goals more concrete)

@dengemann, @DraganaMana, @Izem0, @virvw, @biancasama

### CODE REVIEW

**Comment by massich** [comment]
Here is how it looks the website https://11003-1301584-gh.circle-artifacts.com/0/dev/index.html

and you can edit it through the github website here https://github.com/massich/mne-python/blob/neurospin_workflow_doc/doc/workflows.rst

I'll give you rights to it

**Comment by massich** [comment]
https://11018-1301584-gh.circle-artifacts.com/0/dev/index.html

**Comment by larsoner** [comment]
cc @drammock you would probably be interested in this sort of thing.

@massich @dengemann what is the overall goal / plan here?

**Comment by larsoner** [comment]
This seems to revert (?) some of the changes we made with the last doc revamp, so @choldgraf might also be interested. But really it would be good to hear the plan, and converge on what it should be, before getting too far in the implementation effort (unless people don't mind undoing and redoing a bunch of work)

**Comment by drammock** [comment]
@massich FYI, I also just started working on some docs improvements, mostly focused on fleshing out some of the missing details in the tutorials for new users.  So I'm also keen to hear more about the overall vision behind this reorganization, so we don't duplicate efforts.

**Comment by agramfort** [comment]
let me try to clarify what we want to achieve here. DIsclaimer it's still a bit fuzzy in our heads.

Reports from user is that the documentation page contains too much information and the getting started bar is still too high.

Proposals:

- Start the documentation page with 3-4 paragraph that is some kind of preface to the documentation saying what will be covered, what you will learn and why?
- Put some 2 levels of readings: highlights the tutorials and pages that are really a must read (eg with an icon), and make the rest more advanced. @dengemann was saying that it would like to see immediately what are the 4-5 tutorials that you must read ie the ones we typically teach during an mne-python course.

**Comment by jona-sassenhagen** [comment]
*Something like this* could potentially be very important for newbies.

**Comment by agramfort** [comment]
here is an image from @LauriParkkonen that could be useful to adapt / integrate

source_modelling_workflow

**Comment by dengemann** [comment]
>This seems to revert (?) some of the changes we made with the last doc revamp, so @choldgraf might also be interested. But really it would be good to hear the plan, and converge on what it should be, before getting too far in the implementation effort (unless people don't mind undoing and redoing a bunch of work)

@larsoner @drammock @choldgraf this is really WIP and -- as @agramfort said -- is to help us clarify coming up with clean proposal. 

I think we need to have one clear and simple entry-point into the docs, where there is only 1 recommended reading path, such that newcomers clearly see what are the first materials they should read in which order. Also this should simplify teaching by having the essential notebooks and tutorials flat in one document that is not buried in the depths of our website and that requires no clicking on toggle boxes.

**Comment by jona-sassenhagen** [comment]
> here is an image from @LauriParkkonen that could be useful to adapt / integrate
> 
> source_modelling_workflow

Yes, match it up with links to functioning MNE code pieces, the relevant papers, and maybe also a discussion how to report it and that would be a great piece.

**Comment by drammock** [comment]
> I think we need to have one clear and simple entry-point into the docs, where there is only 1 recommended reading path, such that newcomers clearly see what are the first materials they should read in which order. Also this should simplify teaching by having the essential notebooks and tutorials flat in one document that is not buried in the depths of our website and that requires no clicking on toggle boxes.

I like the sound of that.

FWIW, the rough outline I've been working on is so far something like this:

```
# getting started with MNE-python (high-level overview)
- loading (different readers, the `Raw` object)
- important concepts (`.info`, `events`, `Annotations`)
- epoching (`Epochs` object)
- averaging (`Evoked` object)
- estimating cortical activity (`SourceEstimate` object)
- plotting (raw, epochs, evoked, source)
- command-line tools

# working with continuous data
- loading example data
- querying sample rate, channels, times
- setting EEG reference
- subselecting data
    - selecting channels (`mne.pick_types()`, `mne.pick_channels()`)
    - selecting time spans (`raw.times`, `raw.time_as_index()`)
- dropping and cropping (`raw.drop_channels()`, `raw.crop()`)
- concatenation (`raw.append([...])`)
- plotting (`raw.plot()`, `raw.plot_psd()`)

# working with sensor locations
- montages and layout files (`raw.set_montage()`, `mne.channels.read_layout()`, `mne.channels.read_dig_montage()`, etc)
- plotting sensor locations (`raw.plot_sensors()` with/without `kind='3d'`, `mne.viz.plot_layout()` `mne.viz.plot_alignment()`)

# preprocessing Raw data
- filtering
- resampling
- annotating bad channels / spans
- interpolating bad channels (?)
- artifact rejection
- dimensionality reduction

# epoching raw data
- event-related epoching (finding events, tmin / tmax, baseline correction, selecting / indexing epochs with single index vs. slice, continuous epoching)
- epoch rejection (threshold-based, annotation-based, drop logs, `epochs.selection`)
- plotting epochs
- adding metadata / pandas / dataframes
- saving epochs

# working with averaged data
- making `Evoked` from `Epochs`
- loading `Evoked` from file
- `evoked.data`, `evoked.ch_names`
- plotting
- basic statistical test (condition A / condition B)
- saving

# working with projections
- what are they?
- how to get/set (in the `info` objs, methods like `add_proj`)
- when to apply projections?
- plotting 
    - plotting data with/without projections
    - `raw.plot_projs_topomap()`
```

**Comment by massich** [comment]
@drammock do you have a branch or something? somewhere so that we can check it out?
is this an entry page orthogonal to what was in documentation?

**Comment by drammock** [comment]
@massich none of my outline is implemented yet, it's so far just a plan in progress.  The idea is that it would refine or replace portions of the current documentation, so no, not an orthogonal entry point.  I haven't thought very hard about some things (like whether to keep collapsible sections, for example; personally I don't like them but I haven't fully thought through the pros/cons of keeping them).

**Comment by massich** [comment]
> (like whether to keep collapsible sections, for example; personally I don't like them but I haven't fully thought through the pros/cons of keeping them).

yep, I don't really like it either. When I was working on forward-inverse i had to click to a bunch of places every time to get where I wanted 'cos the menus where folding.

**Comment by massich** [comment]
(some what related, does anyone know how could we easily sort the auto-examples https://github.com/massich/mne-python/pull/20)

**Comment by agramfort** [comment]
maybe https://sphinx-gallery.readthedocs.io/en/latest/gen_modules/sphinx_gallery.sorting.html
?

File: doc/workflows.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+.. .. contents:: Contents
+..    :local:
+
+.. currentmodule:: mne
+
+The documentation in mne-python provides information in the form of: 
+•	Workflows
+•	Tutorials
+•	Example codes
```
Comment (mmagnuski) [inline doc/workflows.rst]: please use singular form "code", code is uncountable :)

**Comment by massich** [comment]
@drammock what are your thoughts here? shall we just close?

**Comment by drammock** [comment]
@massich as far as I'm concerned this one can be closed... I added your fork as a remote, so I can keep in mind the changes suggested here.  I'll make sure to ping you when I open the next docs PR, which will focus on a single entry point to the (post-installation) docs, with a defined sequential ordering of pages, and more narrative explanations.

**Comment by massich** [comment]
It was not me who was driving the changes here. This was collective work done in a hackaton. We should keep it in mind, but not lose our head about it. Things are moving, and they will consolidate in some form or another.

@drammock what I think is important is this part

```
    'subsection_order': ExplicitOrder(['../examples/io/',
                                       '../examples/simulation/',
                                       '../examples/preprocessing/',
                                       '../examples/visualization/',
                                       '../examples/time_frequency/',
                                       '../examples/stats/',
                                       '../examples/decoding/',
                                       '../examples/connectivity/',
                                       '../examples/forward/',
                                       '../examples/inverse/',
                                       '../examples/realtime/',
                                       '../examples/datasets/',
                                       '../tutorials/']),
```

I'll open a PR with it.

**Pull request closed without merging**